lib/plugins/aws/deploy/compile/events/apiGateway/lib/validate.js

```diff
@@ -5,9 +5,9 @@ const BbPromise = require('bluebird');
 
 module.exports = {
   validate() {
-    if (!this.serverless.service.resources.Resources) {
-      throw new this.serverless.classes.Error(
-        'This plugin needs access to Resources section of the AWS CloudFormation template');
+    if (!this.serverless.service.provider.compiledCloudFormationTemplate) {
```

Comment:
Imho this validation can be dropped. Don't think we need this anywhere as we make sure the core template is in place before